api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
behavior of change 
This is what I observed after writing a few tests.
Before the change:-
on voiding and unvoiding an obs without any change no new obs is created.
On voiding an obs with change and then unvoiding it, the change is not considered and thus yet again no new obs is created and the same obs gets unvoided. The same thing happens when an obs without any change is voided and then unvoided with changes.
After the change:- 
on voiding and unvoiding an obs without any change no new obs is created.
On voiding an obs with change and then unvoiding it, the change is now considered, and hence a new obs is created on unvoiding and this change is carried to the groupmembers of the obs. Same thing happens when an obs is voided and then unvoided with changes.
Hence as @wluyima said  there is obscure behavior. How should i proceed with the ticket ?
(Here are the test that i added to ObsServiceTest.java http://pastebin.com/J6FTY1rx, please check and let me know whether the test are wrong) :)